### DISCUSSION TITLE
Support for HMC5883L

### DISCUSSION THREAD

**Discussion by tronikos** [tronikos/esphome-magnetometer-water-gas-meter#11] [General]
@adroush and @sorz2122 I noticed you forked this repo to add support for HMC5883L. I added support myself in a different, IMO easier, way without requiring a fork. You just need to include `hmc5883l.yaml` which overrides and removes some settings. Can you please test it?

**Comment by sorz2122** [comment]
I will Test it

**Reply by sorz2122** [reply]
@tronikos  Apparently I can't establish a connection with my sensor, it may be defective. I have now ordered 2 pieces of GY-271 QMC5883L from Amazon and am testing them. But I haven't found out yet that they have a temperature sensor with a board like yours?

**Reply by tronikos** [reply]
Per the datasheet all QMC5883L should have temperature sensor.

**Comment by adroush** [comment]
Hi @tronikos, 
I tested your solution and can confirm it works great with the HMC5883L sensor.

Thank you!

**Comment by tronikos** [comment]
Thanks for confirming.
